lib/Applify.pm

```diff
@@ -280,7 +283,13 @@ sub app {
 
   for my $option (@{$self->{options}}) {
     my $switch = $self->_attr_to_option($option->{name});
-    push @options_spec, $self->_calculate_option_spec($option);
+    my @spec   = $self->_calculate_option_spec($option);
+    push @options_spec, $spec[0];
+    push @options_spec, sub {
+      my $ref = \$options{$switch};
+      ref($$ref) ? push @$$ref, $spec[1]->(@_) : ($$ref = $spec[1]->(@_));
+      }
+      if $spec[1];
```

Comment:
I think this is a bit hard to read. I'm thinking it could be easier to do this inside _generate_application_class(), where the accessor is built?